### PR TITLE
test: implement 2 stage upgrade

### PR DESCRIPTION
**Pull request by yangchiu**
test: implement 2 stage upgrade

For https://github.com/longhorn/longhorn/issues/4817

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

**Review by khushboo-rancher — COMMENTED** [review]
Can you also include the param `LONGHORN_TRANSIENT_VERSION` in our `longhorn-tests-regression` job?

File: jenkins-jobs/master/sles/longhorn-2-stage-upgrade-tests-sles-amd64.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          default: "k3s"
+          description: "kubernetes distro version to install [rke, k3s] (default: k3s)"
+      - string:
+          name: K8S_DISTRO_VERSION
+          default: "v1.23.1+k3s2"
+          description: |
+              kubernetes version that will be deployed
+              for rke: (default: v1.22.5-rancher1-1)
+              for rke2: (default: v1.25.3+rke2r1)
+              for k3s: (default: v1.25.3+k3s1)
```
[the remark this answers is not in the document]
Comment (yangchiu) [reply jenkins-jobs/master/sles/longhorn-2-stage-upgrade-tests-sles-amd64.yml]: modified!

**Comment by yangchiu** [comment]
> Can you also include the param `LONGHORN_TRANSIENT_VERSION` in our `longhorn-tests-regression` job?

added!

**Review by khushboo-rancher — APPROVED** [review]

**Review by innobead — APPROVED** [review]
LGTM

**Pull request merged**